lib/iris/experimental/ugrid.py

```diff
@@ -615,6 +615,437 @@ def equal(self, other, lenient=None):
         return super().equal(other, lenient=lenient)
 
 
+# class Mesh(CFVariableMixin):
+#     """
+#
+#     .. todo::
+#
+#     .. questions::
+#
+#         - decide on the verbose/succinct version of __str__ vs __repr__
+#
+#     .. notes::
+#
+#         - the mesh is location agnostic
+#
+#         - no need to support volume at mesh level, yet
+#
+#         - topology_dimension
+#             - use for fast equality between Mesh instances
+#             - checking connectivity dimensionality, specifically the highest dimensonality of the
+#               "geometric element" being added i.e., reference the src_location/tgt_location
+#             - used to honour and enforce the minimum UGRID connectivity contract
```

Comment:
Maybe we could we add a little more clarity on this description, as discussed.

Reply:
https://github.com/SciTools/iris/pull/4005#issuecomment-779174290